opencog/nlp/lojban/HaskellLib/src/OpenCog/Lojban/Syntax/Util.hs

```diff
@@ -273,6 +291,11 @@ getFlagValueIso flag = Iso f g where
 setFlagValueIso :: Flag -> SynIso String ()
 setFlagValueIso flag = inverse (getFlagValueIso flag)
 
+setFlagValueIso2 :: Flag -> String -> SynIso a a
+setFlagValueIso2 flag val = Iso f g
+    where f a = setFlagValue flag val >> pure a
+          g a = rmFlag flag >> pure a
+
```

Comment:
So almost all of your code seems not to contain any comments that explain what it actually does. I don't want to be a pill and explain why writing comments for code is good.  For example, it helps future-you remember what something does. and future-others. And, rarely, having to explain something in comments helps you realize that there's a bettter design.